rep.go

```diff
@@ -7,12 +7,18 @@ package zmq4
 import (
 	"context"
 	"net"
+	"sync"
+
+	"golang.org/x/xerrors"
 )
 
 // NewRep returns a new REP ZeroMQ socket.
 // The returned socket value is initially unbound.
 func NewRep(ctx context.Context, opts ...Option) Socket {
 	rep := &repSocket{newSocket(ctx, Rep, opts...)}
+	sharedState := newRepState()
+	rep.sck.w = newRepWriter(rep.sck.ctx, sharedState)
```

Comment:
shouldn't the `repWriter` and `repReader` be closed in `repSocket.Close` ?
(probably an overlook in `socket.Close` itself...)

Reply:
I'll address this in a separate PR.
thanks again.

Reply:
https://github.com/go-zeromq/zmq4/issues/77

Reply:
Sorry. I somehow missed this comment. Yeah you're right. `socket.Close` should be calling Close on its readers and writers.

Reply:
Alternatively, `Socket.Close` could simply `rmConn` the connections from the pools to prevent the "double-close" problem.